app/utils/performLogin.js

```diff
@@ -82,3 +82,50 @@ export default function* performLogin(
   yield sleep(5000);
   dispatch(returnToSearchPage());
 }
+
+export function* performManualLogin(
+  dispatch: Dispatch,
+  electronUtils: ElectronUtilsRedux,
+  configuration: Configuration
+) {
+  yield sleep(1000);
+
+  // there seems to be a problem with the captcha implementation: https://github.com/google/recaptcha/issues/269
+  yield electronUtils.evaluate(`grecaptcha = undefined`);
+
+  // Check again if user is logged in
+  if (yield electronUtils.elementExists('.sso-login--logged-in')) {
+    dispatch(setLoginStatus(LOGINSTATUS.LOGGED_IN));
+    dispatch(setBotMessage('Bereits eingeloggt', 4000));
+  } else {
+    dispatch(setBotMessage('Anmelden'));
+    yield dispatch(
+      electronRouting(
+        'puppet',
+        'https://www.immobilienscout24.de/geschlossenerbereich/start.html?source=headericon'
+      )
+    );
+    yield sleep(3000);
+
+    yield electronUtils.humanInteraction(async () => {
+      const username = await electronUtils.elementExists('#username');
+      const password = await electronUtils.elementExists('#password');
+      const external = electronUtils.isOnExternalPage();
+      return username || password || external;
+    });
+
+    yield sleep(5000);
+
+    yield electronUtils.humanInteraction(async () => {
+      const exists = await electronUtils.elementExists('.mfa-verify');
```

Comment:
This is cool, would it make sense to add it to the automatic login too, in case someone gets into an MFA procedure?

Reply:
I copied it from there ;)